src/machines/activity_state_machine.rs

```diff
@@ -42,17 +58,155 @@ fsm! {
 
     StartedActivityCancelEventRecorded --(ActivityTaskFailed, on_activity_task_failed) --> Failed;
     StartedActivityCancelEventRecorded
-      --(ActivityTaskCompleted, on_activity_task_completed) --> Completed;
+      --(ActivityTaskCompleted(ActivityTaskCompletedEventAttributes), on_activity_task_completed) --> Completed;
     StartedActivityCancelEventRecorded
       --(ActivityTaskTimedOut, on_activity_task_timed_out) --> TimedOut;
     StartedActivityCancelEventRecorded
       --(ActivityTaskCanceled, on_activity_task_canceled) --> Canceled;
 }
 
-#[derive(thiserror::Error, Debug)]
-pub(super) enum ActivityMachineError {}
+#[derive(Debug, derive_more::Display)]
+pub(super) enum ActivityMachineCommand {
+    #[display(fmt = "Complete")]
+    Complete(Option<Payloads>),
+}
+
+#[derive(Debug, Clone, derive_more::Display)]
+pub(super) enum ActivityCancellationType {
+    /**
+     * Wait for activity cancellation completion. Note that activity must heartbeat to receive a
+     * cancellation notification. This can block the cancellation for a long time if activity doesn't
+     * heartbeat or chooses to ignore the cancellation request.
+     */
+    WaitCancellationCompleted,
+
+    /** Initiate a cancellation request and immediately report cancellation to the workflow. */
+    TryCancel,
+
+    /**
+     * Do not request cancellation of the activity and immediately report cancellation to the workflow
+     */
+    Abandon,
+}
 
-pub(super) enum ActivityCommand {}
+impl Default for ActivityCancellationType {
+    fn default() -> Self {
+        ActivityCancellationType::TryCancel
+    }
+}
+
+/// Creates a new, scheduled, activity as a [CancellableCommand]
```

Comment:
Comment is wrong now, I probably didn't update place you copied it from so that needs to be fixed too

Reply:
Changed to:
/// Creates a new activity state machine and a command to schedule it on the server.